relayer/relays/beacon/store/datastore.go

```diff
@@ -169,8 +174,51 @@ func (s *Store) WriteEntry(attestedSlot, finalizedSlot uint64, attestedStateData
 	return s.storeUpdate(attestedSlot, finalizedSlot, attestedSyncPeriod, finalizedSyncPeriod)
 }
 
+func (s *Store) ListBeaconStates() ([]BeaconState, error) {
+	var response []BeaconState
+
+	query := `SELECT id, attested_slot, finalized_slot, attested_sync_period, finalized_sync_period, attested_state_filename, finalized_state_filename FROM beacon_state ORDER BY attested_slot`
+
+	rows, err := s.db.Query(query)
+	if err != nil {
+		return response, fmt.Errorf("no match found")
+	}
```

Comment:
Do we want to implement a pagination query here?

Reply:
It is limited to 100 states so I thought it is not necessary yet, but will be cool for the future!